### PR TITLE
Add Support for more archive types into the Mod Manager allowing for Smaller Downloads E.G 7zip

### PR DESCRIPTION
**Pull request by barkermn01**
There is also a fix for a UI race condition bug with the download progress bar and the downloadMod variable being nulled.

But the main feature is added SharpCompress so the Mod Manager can support all these types of archive,
"rar", "unrar", "zip", "bzip2", "gzip", "tar", "7zip", "lzip", "gz" this works by using the DownloadFormat property of the Mod catalog entry

### CODE REVIEW

**Comment by barkermn01** [comment]
[Memoria.Patcher.zip](https://github.com/user-attachments/files/16576319/Memoria.Patcher.zip)

Patcher with the launcher supporting the new archive versions

**Comment by snouz** [comment]
testing it

**Comment by snouz** [comment]
There's something missing, do we need a special operation in VSStudio (if so, we need to communicate it to people compiling the code) or did you forget to include a dll or something?

image
(name or namespace of SharpCompress doesn't exist)

**Comment by barkermn01** [comment]
it's supposed to be a standard NuGet package. a build should force it to obtain them.

**Comment by snouz** [comment]
I added the reference to the package so it automatically downloads it and compile.

Currently, my tests show that rar and tar work no problem, but 7zip (both in normal and best compressions) don't work. 

image
image
image
image

But I also notice it tries to decompress form a .zip, so .zip might be hardcoded somewhere.
image

**Comment by barkermn01** [comment]
Thanks for that @snouz i will update as soon as i have time to pick this up again little busy with work and personal life at the moment, as soon as i find some free time i will get looking into it, perhapse a missed a bit in the downloader that is forcing it to save the file as a Zip always.

File: Memoria.Launcher/ModManagerWindow.cs [hunk 1]
```diff
@@ -378,7 +415,8 @@ private void DownloadEnd(Object sender, AsyncCompletedEventArgs e)
                 String downloadingModName = downloadingMod.Name;
                 String path = Mod.INSTALLATION_TMP + "/" + (downloadingMod.InstallationPath ?? downloadingModName);
                 Boolean success = false;
-                if (String.IsNullOrEmpty(downloadingMod.DownloadFormat) || downloadingMod.DownloadFormat == "Zip")
+                String downloadFormatLower = downloadingMod.DownloadFormat ?? "zip".ToLower();
```
Comment (barkermn01) [inline Memoria.Launcher/ModManagerWindow.cs:433]: note for self this is a bug it's missing brackets should be `(downloadingMod.DownloadFormat ?? "zip").ToLower();`

**Review by barkermn01 — COMMENTED** [review]
note to fix a bug

File: Memoria.Launcher/ModManagerWindow.cs (hunk 1 above)
Comment (barkermn01) [reply Memoria.Launcher/ModManagerWindow.cs]: Resolved in [ac97eca](https://github.com/Albeoris/Memoria/pull/711/commits/ac97ecaeb0fe8f835ae1fe729c3883cd9592b377)

**Comment by barkermn01** [comment]
Right fixed the problem with specific archives not extracting, 

Also completed the request from @snouz on discord to enable delete key to ask to uninstall the selected mod

**Comment by snouz** [comment]
Tested and both features seem to work correctly.

**Pull request merged**